packages/router/api.js

```diff
@@ -0,0 +1,75 @@
+const express = require('express');
+const router = express.Router();
+const { pathMappingsData, loadPathMappings } = require('./pathmapping');
+const fsp = require("fs").promises;
+const path = require('path');
+const config = require("./config");
+
+const swaggerUi = require('swagger-ui-express');
+const YAML = require('yamljs');
+const openApiDocumentation = YAML.load('./openapi.yaml');
+
+
+
+router.get('/pathMappings', (req, res) => {
+    if (pathMappingsData.pathMappings.length === 0) {
+        res.json({ message: 'No mapping found' });
+    } else {
+        res.json(pathMappingsData.pathMappings);
+    }
+});
+
+
+
+router.get('/pathMappings/reload', async(req, res) => {
+    await loadPathMappings();
+    if (pathMappingsData.pathMappings.length === 0) {
+        res.json({ message: 'No mapping found' });
+    } else {
+        res.json(pathMappingsData.pathMappings);
+    }
+});
+
+
+
+router.post('/pathMappings', (req, res) => {
+    const { incoming_path, mapped_with } = req.body;
+    const index = pathMappings.findIndex(mapping => mapping.incoming_path === incoming_path);
+    if (index !== -1) {
+        // Modify existing entry
+        pathMappingsData.pathMappings[index].mapped_with = mapped_with;
+    } else {
+        // Add new entry
+        pathMappingsData.pathMappings.push({ incoming_path, mapped_with });
+    }
+    res.json(pathMappingsData.pathMappings);
+});
+
+
+router.put('/pathMappings', async (req, res) => {
+    const filePath = path.join(config.get("webroot"), '.routemapping');
+    try {
+        await fsp.writeFile(filePath, JSON.stringify(pathMappingsData.pathMappings), 'utf8');
+        res.json({ message: 'Path mappings successfully written to .routemapping file' });
+    } catch (err) {
+        res.status(500).json({ message: `Error writing to .routemapping file: ${err}` });
+    }
+});
+
+
+router.delete('/pathMappings/:incoming_path', (req, res) => {
```

Comment:
Why is an API written with camel case?

- https://developers.google.com/search/docs/crawling-indexing/url-structure?hl=en&visit_id=638512249754743989-1528210015&rd=1
- https://metamug.com/article/rest-api-naming-best-practices.html#word-delimiters
- https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/

cc @sayak-sarkar